routes/dynamic_indicators.py

```diff
@@ -0,0 +1,895 @@
+    metadata = await describe_via_wcps(cov_id)
+    return metadata
+
+
+def validate_latlon_and_reproject_to_epsg_3338(lat, lon, variable):
+    """Validate lat/lon, then reproject to EPSG:3338"""
+    lat = float(lat)
+    lon = float(lon)
+    if not latlon_is_numeric_and_in_geodetic_range(lat, lon):
+        return render_template("400/bad_request.html"), 400
+
+    var_coverages = all_coverages.get(variable)
+
+    for cov_id in var_coverages:
+
+        metadata = asyncio.run(get_cmip6_metadata(cov_id))
+        cmip6_downscaled_bbox = construct_latlon_bbox_from_coverage_bounds(metadata)
+        within_bounds = validate_latlon_in_bboxes(
+            lat, lon, [cmip6_downscaled_bbox], [cov_id]
+        )
+        if within_bounds == 404:
+            return (
+                render_template("404/no_data.html"),
+                404,
+            )
+        if within_bounds == 422:
+            return (
+                render_template(
+                    "422/invalid_latlon_outside_coverage.html",
+                    bboxes=[cmip6_downscaled_bbox],
+                ),
+                422,
+            )
+
+    lon, lat = project_latlon(lat, lon, dst_crs=3338)
+
+    return lon, lat
+
+
+def validate_operator(operator):
+    """Validate operator is 'above' or 'below' and convert to '>' or '<'"""
+    if operator not in ["above", "below"]:
+        return render_template("400/bad_request.html"), 400
+    if operator == "above":
+        operator = ">"
+    else:
+        operator = "<"
+    return operator
+
+
+def validate_units_threshold_and_variable(units, threshold, variable):
+    """Validate units and threshold based on variable type. Convert threshold to standard units if needed."""
+    if variable in ["tasmax", "tasmin"]:
+        if units not in ["C", "F"]:
+            raise ValueError("Units for temperature must be 'C' or 'F'.")
+        if threshold is not None:
+            if units == "F":
+                threshold = (float(threshold) - 32) * 5.0 / 9.0
+            else:
+                threshold = float(threshold)
+    elif variable == "pr":
+        if units not in ["mm", "in"]:
+            return render_template("400/bad_request.html"), 400
+        if threshold is not None:
+            if units == "in":
+                threshold = float(threshold) * 25.4
+            else:
+                threshold = float(threshold)
+    else:
+        return render_template("400/bad_request.html"), 400
+
+    # precipitation thresholds should be >= 0
+    if variable == "pr" and threshold is not None and threshold < 0:
+        return render_template("400/bad_request.html"), 400
+
+    return units, threshold
+
+
+def validate_stat(stat):
+    """Validate that stat is one of 'max', 'min', 'mean', or 'sum'."""
+    if stat not in ["max", "min", "mean", "sum"]:
+        return render_template("400/bad_request.html"), 400
+    if stat == "mean":
+        stat = "avg"  # NOTE: rasdaman uses 'avg' instead of 'mean' in WCPS queries
```

Comment:
If rasdaman wants "avg" why not just use that, instead of mapping "mean" to "avg"?